### PR TITLE
web: add trace view

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu, @maiamcc,

Please review the following commits I made in branch nicks/buildview:

9c11fa394b63cf6c658cc41102c3642952d206d6 (2020-02-20 16:32:53 -0500)
web: add trace view

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
here's what it looks like:

Screenshot from 2020-02-20 16-33-24

There's currently no way to get to this in the UI. You have to know the magic URL. 

It tries to visualize all the logs that were "caused" by a particular build. 

Future directions:
- A button to get to this UI
- Light affordances to express "historic" builds (with a clear start/end time) vs pending/ongoing builds
- breadcrumbs to navigate back to the full log, or to previous/future builds

**Comment by nicks** [comment]
ping?

File: web/src/SecondaryNav.tsx [hunk 1]
```diff
@@ -71,6 +73,36 @@ class SecondaryNav extends PureComponent<NavProps> {
     let logIsSelected = this.props.resourceView === ResourceView.Log
     let alertsIsSelected = this.props.resourceView === ResourceView.Alerts
     let facetsIsSelected = this.props.resourceView === ResourceView.Facets
+    let traceIsSelected = this.props.resourceView === ResourceView.Trace
+    let spanId = this.props.span
+
+    let traceItem = null
+    if (traceIsSelected && this.props.traceUrl) {
```
Comment (maiamcc) [inline web/src/SecondaryNav.tsx]: add a note here that this is to keep this tab hidden unless you know the url / `if traceIsSelected` should be removed when we make this discoverable, etc.

Comment (nicks) [reply web/src/SecondaryNav.tsx]: done

**Comment by maiamcc** [comment]
Thanks for the link. FWIW that doc doesn't precisely answer my question as to the variable name "trace" in particular (the word "trace" doesn't actually appear in the doc?) but I don't feel strongly. just relevant info for you that that variable name isn't immediately legible, do with that what you will

**Pull request merged**